src/core/sock/sock-redirect.cpp

```diff
@@ -1103,12 +1105,14 @@ EXPORT_SYMBOL int XLIO_SYMBOL(getsockopt)(int __fd, int __level, int __optname,
 
     srdr_logdbg_entry("fd=%d, level=%d, optname=%d", __fd, __level, __optname);
 
+#ifndef XLIO_STATIC_BUILD
```

Comment:
What happens with xlio_get_api() function which can be used from <xlio_extra.h> even with static linkage? I expect it will fail with static lib, because getsockopt() will go to kernel?

I guess not much we can do with current xlio_get_api() implementation...

Reply:
With the static library, there is no reason to use `getsockopt` with `SO_XLIO_GET_API`.